python/paddle/fluid/layers/nn.py

```diff
@@ -6355,10 +6355,33 @@ def sampled_softmax_with_cross_entropy(logits,
         })
     loss = helper.create_variable_for_type_inference(dtype=logits.dtype)
     softmax = helper.create_variable_for_type_inference(dtype=logits.dtype)
+
+    one_hot_input = {'X': sampled_label}
+    one_hot_attr = {}
+    if in_dygraph_mode():
+        one_hot_attr['depth'] = num_samples + 1
+    else:
+
+        depth_var = helper.create_variable_for_type_inference(dtype='int32')
+        helper.append_op(
+            type='fill_constant',
+            inputs={},
+            outputs={'Out': [depth_var]},
+            attrs={
+                'shape': [1],
+                'dtype': depth_var.dtype,
+                'value': float(num_samples + 1),
+                'force_cpu': True,
+            },
+            stop_gradient=True)
+        depth_var.stop_gradient = True
+
+        one_hot_input['depth_tensor'] = depth_var
```

Comment:
What's the meaning of line 6365~6379?
Why do you change the origin implementation?